### PR TITLE
Fix slow bn test skip

### PR DESCRIPTION
**Pull request by dnovakovic-dxc**
## Proposed changes

Some BN tests were taking too much time just to be skipped if not supported (several tens of seconds). Moved `GTEST_SKIP` part at the beginning of the `SetUp` method in `bn.hpp` test file to skip the test execution immediately if needed.

## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [ x ] I have added automated tests relevant to the introduced functionality
- [ x ] I have sufficient test coverage for the changes, and code coverage hasn't decreased as a result of my PR
- [ x ] I have ran the tests, and they are all passing locally
- [ x ] I have added relevant documentation for the changes
- [ x ] I have removed the stale documentation which is no longer relevant after this pull request
- [ x ] I have ran `make format` & `make check_format` to ensure the changes have been formatted

### CODE REVIEW

File: test/gtest/bn.hpp [hunk 1]
```diff
@@ -119,19 +119,19 @@ struct BNInferTest : public ::testing::TestWithParam<std::tuple<TestCase,
 protected:
     void SetUp() override
     {
+        auto&& handle = get_handle();
+        if(!miopen::solver::ck_utility::is_ck_whitelist(handle.GetStream()))
```
Comment (bghimireamd) [inline test/gtest/bn.hpp:123]: but batch norm does not have restriction on gpu arch no? https://github.com/ROCm/MIOpen/blob/develop/src/solver/batchnorm/forward_inference.cpp#L41

Comment (dnovakovic-dxc) [reply test/gtest/bn.hpp:123]: Okay yes, I see from this PR https://github.com/ROCm/MIOpen/pull/3888 that this `is_ck_whitelist` check will be removed completely, I will close this PR and changes can be applied from the mentioned one
